catalyst/dl/callbacks/optimizer.py

```diff
@@ -73,26 +81,36 @@ def on_epoch_start(self, state):
         for i in range(len(optimizer.param_groups)):
             safitty.set(optimizer.param_groups, i, "weight_decay", value=0.0)
 
-    def on_batch_start(self, state):
-        state.loss = None
-
-    def on_batch_end(self, state):
+    def _get_loss(self, state) -> torch.Tensor:
         loss = state.get_key(key="loss", inner_key=self.loss_key)
-        if isinstance(loss, dict):
-            loss = list(loss.values())
-        if isinstance(loss, list):
-            loss = torch.mean(torch.stack(loss))
 
-        if self.prefix is not None:
-            state.metrics.add_batch_value(
-                metrics_dict={
-                    self.prefix: loss.item(),
-                }
+        if isinstance(loss, list):
+            raise ValueError(
+                f"Loss is a list. "
+                f"Only the last value will be used for `backward`."
+                f"To aggregate losses into "
+                "one value use `CriterionAggregatorCallback`"
             )
+        if isinstance(loss, dict):
+            if len(loss) > 1:
+                error = "Loss is a dict, to aggregate losses into " \
+                        "one value use `CriterionAggregatorCallback`."
+                if self.loss_key is None:
+                    error = error + " Or try to pass `loss_key` " \
+                                    "in the OptimizerCallback init"
+                raise ValueError(error)
+            loss = list(loss.values())[-1]
+        return loss
```

Comment:
why do you throw an exception for List case and take last loss for dict?
I suggest to remove this method and use only
```python
loss = state.get_key(key="loss", inner_key=self.loss_key)
```
all aggregation logic should be done in another callback